src/cuga/demo_tools/crm/crm_api/leads.py

```
@@ -1,38 +1,40 @@
 @router.get("/{lead_id}", response_model=LeadResponse)
-def get_lead(lead_id: int, db: Session = Depends(get_db)):
+def get_lead(lead_id: int, db: sqlite3.Connection = Depends(get_db)):
     lead = lead_crud.get(db, lead_id)
     if not lead:
         raise HTTPException(status_code=404, detail="Lead not found")
     return lead
 
 
 @router.put("/{lead_id}", response_model=LeadResponse)
-def update_lead(lead_id: int, lead: LeadUpdate, db: Session = Depends(get_db)):
+def update_lead(lead_id: int, lead: LeadUpdate, db: sqlite3.Connection = Depends(get_db)):
```

Ruff (0.15.11)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)